src/functions/newsletter-submission.js

```diff
@@ -0,0 +1,51 @@
+const fetch = require("node-fetch");
+const { BUTTONDOWN_API_KEY } = process.env;
+
+const config = (userEmail) => {
+  return {
+    method: "POST",
+    headers: {
+      Authorization: `Token ${BUTTONDOWN_API_KEY}`,
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({ email: userEmail }),
+  };
+};
+
+exports.handler = async ({ body }) => {
+  const email = getEmail(body);
+
+  try {
+    const response = await fetch(
+      "https://api.buttondown.email/v1/subscribers",
+      config(email)
+    );
+
+    const data = await response.json();
+    if (!hasError(data)) {
+      return respondWith(400, "Could not sign user up to newsletter");
```

Comment:
Since these are the messages we'll display to the user, they should talk to them, not about them. "We we're able to subscribe you. Please try again." Something like that. This one and the 500 below. That one can be something like; "We ran into an error. Please try again."